File: src/main/java/com/ambition/business/user/service/impl/SysRolePermissionServiceImpl.java
package com.ambition.business.user.service.impl;

import com.ambition.business.user.domain.SysRolePermission;
import com.ambition.business.user.mapper.SysRolePermissionMapper;
import com.ambition.business.user.service.ISysRolePermissionService;
import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import com.ambition.common.constants.Constants;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.stereotype.Service;
import com.ambition.common.util.R;
import com.baomidou.mybatisplus.core.metadata.IPage;
import com.baomidou.mybatisplus.core.toolkit.Wrappers;
import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import java.util.List;
import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;

/**
 * <p>
 * 角色-权限关联表 服务实现类
 * </p>
 *
 * @author wuys
 * @since 2019-12-11
 */
@Service
public class SysRolePermissionServiceImpl extends ServiceImpl<SysRolePermissionMapper, SysRolePermission> implements ISysRolePermissionService {

    @Override
    @Transactional
    public R findListByPage(int page, int pageSize,Long groupId){
     if (pageSize == 0) {
        pageSize = Constants.DEFAULT_PAGESIZE;
     }
     Page<SysRolePermission> dictPage = new Page<>(page, pageSize);
     IPage<SysRolePermission> sysDictIPage = null;
     LambdaQueryWrapper<SysRolePermission> lambdaQueryWrapper = Wrappers.<SysRolePermission>lambdaQuery();
     sysDictIPage = baseMapper.selectPage(dictPage, lambdaQueryWrapper.select(SysRolePermission.class, i -> true));
     return R.ok(sysDictIPage);
    }

    @Override
    public R deleteByIds(List<Long> ids){
     int result = this.baseMapper.deleteBatchIds(ids);
     if(result > 0){
        return R.ok();
     }
     return R.error();
    }

    @Override
    public R getById(Long id){
      SysRolePermission e = super.getById(id);
      return R.ok(e);
    }

    @Override
    public R saveSysRolePermission(SysRolePermission entity){
      boolean r = this.save(entity);
      if(r){
        return R.ok();
       }
      return R.error();
    }

    @Override
    public R updateSysRolePermissionById(SysRolePermission entity){
        boolean r = this.updateById(entity);
        if(r){
            return R.ok();
        }
        return R.error();
    }

}

File: src/main/java/com/ambition/business/user/service/impl/SysRolePermissionServiceImpl.kt
package com.ambition.business.user.service.impl

import com.ambition.business.user.domain.SysRolePermission
import com.ambition.business.user.mapper.SysRolePermissionMapper
import com.ambition.business.user.service.ISysRolePermissionService
import com.ambition.common.constants.Constants
import com.ambition.common.util.R
import com.baomidou.mybatisplus.core.metadata.IPage
import com.baomidou.mybatisplus.core.toolkit.Wrappers
import com.baomidou.mybatisplus.core.toolkit.support.SFunction
import com.baomidou.mybatisplus.extension.plugins.pagination.Page
import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.stream.Collectors

/**
 * <pre>
 *
 *
 * @title: SysRolePermissionServiceImpl
 * @description:
 * @company: 润投科技
 * @author: wuys
 * @datetime: 2019-12-18 16:09
 * </pre>
 */
@Service
open class SysRolePermissionServiceImpl : ServiceImpl<SysRolePermissionMapper, SysRolePermission>(), ISysRolePermissionService {

    override fun getPermissionsByRoleId(roleId: Int?): List<Int> {
        val sysRolePermissions = baseMapper.selectList(Wrappers.lambdaQuery<SysRolePermission>().eq(SFunction<SysRolePermission, Any> { SysRolePermission::roleId }, roleId))
        return sysRolePermissions.stream().map { sysRolePermission -> sysRolePermission.permissionId }
                .collect(Collectors.toList<Int>())
    }

    @Transactional
    override fun findListByPage(page: Int, pageSize: Int, groupId: Long?): R {
        var pageSize = pageSize
        if (pageSize == 0) {
            pageSize = Constants.DEFAULT_PAGESIZE
        }
        val dictPage = Page<SysRolePermission>(page.toLong(), pageSize.toLong())
        var sysDictIPage: IPage<SysRolePermission>? = null
        val lambdaQueryWrapper = Wrappers.lambdaQuery<SysRolePermission>()
        sysDictIPage = baseMapper.selectPage(dictPage, lambdaQueryWrapper.select(SysRolePermission::class.java) { i -> true })
        return R.ok(sysDictIPage)
    }

    override fun deleteByIds(ids: List<Long>): R {
        val result = this.baseMapper.deleteBatchIds(ids)
        return if (result > 0) {
            R.ok()
        } else R.error()
    }

    override fun getById(id: Long?): R {
        val e = super.getById(id)
        return R.ok(e)
    }

    override fun saveSysRolePermission(entity: SysRolePermission): R {
        val r = this.save(entity)
        return if (r) {
            R.ok()
        } else R.error()
    }

    override fun updateSysRolePermissionById(entity: SysRolePermission): R {
        val r = this.updateById(entity)
        return if (r) {
            R.ok()
        } else R.error()
    }

}